dl/src/main/scala/com/intel/analytics/bigdl/python/api/PythonBigDL.scala

```diff
@@ -242,6 +242,664 @@ class PythonBigDL[T: ClassTag](implicit ev: TensorNumeric[T]) extends Serializab
+  : Normalize[T] = {
+    Normalize[T](p,
+      eps)
+  }
+
+  def createPReLU(nOutputPlane: Int = 0)
+  : PReLU[T] = {
+    PReLU[T](nOutputPlane)
+  }
+
+  def createPadding(dim: Int,
+                    pad: Int,
+                    nInputDim: Int,
+                    value: Double = 0.0,
+                    nIndex: Int = 1)
+  : Padding[T] = {
+    Padding[T](dim,
+      pad,
+      nInputDim,
+      value,
+      nIndex)
+  }
+
+  def createPairwiseDistance(norm: Int = 2)
+  : PairwiseDistance[T] = {
+    PairwiseDistance[T](norm)
+  }
+
+  def createParallelCriterion(repeatTarget: Boolean = false)
+  : ParallelCriterion[T] = {
+    ParallelCriterion[T](repeatTarget)
+  }
+
+  def createParallelTable()
+  : ParallelTable[T] = {
+    ParallelTable[T]()
+  }
+
+  def createPower(power: Double,
+                  scale: Double = 1,
+                  shift: Double = 0)
+  : Power[T] = {
+    Power[T](power,
+      scale,
+      shift)
+  }
+
+  def createRReLU(lower: Double = 1.0 / 8,
+                  upper: Double = 1.0 / 3,
+                  inplace: Boolean = false)
+  : RReLU[T] = {
+    RReLU[T](lower,
+      upper,
+      inplace)
+  }
+
+  def createReLU6(inplace: Boolean = false)
+  : ReLU6[T] = {
+    ReLU6[T](inplace)
+  }
+
+  def createReplicate(nFeatures: Int,
+                      dim: Int = 1,
+                      nDim: Int = Int.MaxValue)
+  : Replicate[T] = {
+    Replicate[T](nFeatures,
+      dim,
+      nDim)
+  }
+
+  def createRoiPooling(pooled_w: Int, pooled_h: Int, spatial_scale: T)
+  : RoiPooling[T] = {
+    RoiPooling[T](pooled_w,
+      pooled_h,
+      spatial_scale)
+  }
+
+  def createScale(size: JList[Int])
+  : Scale[T] = {
+    Scale[T](size.asScala.toArray)
+  }
+
+  def createSelect(dimension: Int,
+                   index: Int)
+  : Select[T] = {
+    Select[T](dimension,
+      index)
+  }
+
+  def createSelectTable(dimension: Int)
+  : SelectTable[T] = {
+    SelectTable[T](dimension)
+  }
+
+  def createSigmoid()
+  : Sigmoid[T] = {
+    Sigmoid[T]()
+  }
+
+  def createSmoothL1Criterion(sizeAverage: Boolean = true)
+  : SmoothL1Criterion[T] = {
+    SmoothL1Criterion[T](sizeAverage)
+  }
+
+  def createSmoothL1CriterionWithWeights(sigma: Double, num: Int = 0)
+  : SmoothL1CriterionWithWeights[T] = {
+    SmoothL1CriterionWithWeights[T](sigma,
+      num)
+  }
+
+  def createSoftMax()
+  : SoftMax[T] = {
+    SoftMax[T]()
+  }
+
+  def createSoftMin()
+  : SoftMin[T] = {
+    SoftMin[T]()
+  }
+
+  def createSoftPlus(beta: Double = 1.0)
+  : SoftPlus[T] = {
+    SoftPlus[T](beta)
+  }
+
+  def createSoftShrink(lambda: Double = 0.5)
+  : SoftShrink[T] = {
+    SoftShrink[T](lambda)
+  }
+
+  def createSoftSign()
+  : SoftSign[T] = {
+    SoftSign[T]()
+  }
+
+  def createSoftmaxWithCriterion(ignoreLabel: Option[Int] = None,
+                                 normalizeMode: String = "VALID")
+  : SoftmaxWithCriterion[T] = {
+    val normM = normalizeMode match {
```

Comment:
A user friendly message, ie:  Invalid normalizeMode : $normalizeMode , we only support FULL, BATCH_SIZE and NONE for now.